### DISCUSSION TITLE
Why is generic type argument non-nullable (causing warning) when null is passed to params T[]?

### DISCUSSION THREAD

**Discussion by jnm2** [dotnet/csharplang#5269] [Q&A] [answered]
This is counterintuitive behavior which makes it necessary to explicitly specify the type argument with `ImmutableArray.Create`. 

[(SharpLab)](https://sharplab.io/#v2:C4LgbgpgTglgZgTwD4GIB2BXANlghgIywgAII0CiBYAKBoFkAKTHAGmICJgIBnYdgSgDcNGgAEALMToAeACoA+BgAdcUXAFtuxWQG0AusTC4sGHv2IBvYgF8aQA=)
```cs
#nullable enable

// ⚠ CS8625 Cannot convert null literal to non-nullable reference type.
M(null, "test");

void M<T>(params T[] values) { }
```

**Comment by RikkiGibson** [comment, marked as answer]
This looks like a scenario that was fixed by dotnet/roslyn#56022. No warning is given when using the "main" branch in [SharpLab](https://sharplab.io/#v2:EYLgtghglgdgNAFxANwKYCcoDMCeAfAYhgFcAbUiYU1AAlRkuoFgAoVgWQAoTy4aAiBKgDOCfgEoA3K1YABACw12AHgAqAPk4AHCOghhhNVQG0AujWQRSxEeJoBvGgF9WQA=).
